src/app/modules/candidate-registration/components/registration-form/registration-form.component.ts

```diff
@@ -81,10 +83,18 @@ export class RegistrationFormComponent implements OnInit {
     public candidateRegistrationService: CandidateRegistrationService,
     private toastr: ToastrService,
     private loadingService: LoadingService,
+    private router: Router,
   ) {}
 
   ngOnInit(): void {
     this.createRegistrationForm();
+
+    if (this.router.url.includes('/edit')) {
+      const details = window?.history?.state['candidateDetails'] || {};
```

Comment:
Add like `(window?.history?.state || window?.history?.state['candidateDetails']) || {};` else it will fail test cases, I just fixed in `QuizTestcomponent`